HeterogeneousCore/SonicTriton/interface/TritonData.h

```diff
@@ -132,11 +132,12 @@ class TritonData {
   std::shared_ptr<void> holder_;
   std::shared_ptr<TritonMemResource<IO>> memResource_;
   std::shared_ptr<Result> result_;
+  mutable bool done_{};
```

Comment:
Why is this `mutable`?

Reply:
Ok, I see below why. Please add a comment for the `TritonData` class that it is not (intended to be?) const-thread-safe.

Alternatively consider making the `fromServer()` method non-const. Is the requirement "can be called only once per event" only a sanity check for the logic using `TritonData`, or is there a deeper reason?

Reply:
The output object is given to the `produce()` function as const, because the user shouldn't modify it. (So the method has to be const in order to be callable in that context.) SONIC always uses stream modules (or one modules for analyzers), never global modules.

Where would you prefer that I put the comment about const-thread-safe?

Allowing it to be called multiple times per event would require a significant change in the current logic, and it would be inefficient anyway. I think it's better to enforce the better pattern.

Reply:
I'd add at the top of the class definition
https://github.com/cms-sw/cmssw/blob/2fda345fb53345d39b5dd2d28535e0c77c82cdcc/HeterogeneousCore/SonicTriton/interface/TritonData.h#L41-L43

Some short comment here would actually be helpful as well, accompanied with the `CMS_SA_ALLOW` decoration
```suggestion
  CMS_SA_ALLOW mutable bool done_{};
```

Reply:
@makortel I added the recommended comments, and also added a more explicit comment about another non-const-thread-safe behavior in `fromServer()` that was introduced in the previous #33801